src/Model/PhasorDynamics/SystemModel.hpp

```diff
@@ -648,10 +510,89 @@ namespace GridKit
         components_.push_back(component);
       }
 
+      void update() override
+      {
+        updateChildrenAdjoint();
+      }
+
     private:
       std::vector<bus_type*>       buses_;
       std::vector<component_type*> components_;
 
+      /// @brief Whenever the internal state of a system is updated, the system
+      ///        needs to reflect those changes in its "children" aka buses and
+      ///        components. `updateChildren` will only update `y`, `yp`, and
+      ///        `param` - call when you know only the internal state of those
+      ///        variables has been updated.
+      void updateChildren()
+      {
+        // Update variables
+        IdxT varOffset = 0;
+        IdxT optOffset = 0;
```

Comment:
This is very helpful! I suggest we put this in a separate PR as it is out of scope here.

Reply:
Just the `updateChildren()` function or the entire 37ea7213041afef6ac1e9ce85855db1c83bef53f commit?

Reply:
Yes, just `updateChildren()`. It has little to do with callbacks.